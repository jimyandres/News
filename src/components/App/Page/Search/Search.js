import React, { Component } from 'react';
import PropTypes from 'prop-types';
import TextField from 'material-ui/TextField';
import RaisedButton from 'material-ui/Button';
import './index.css';

class Search extends Component {
  componentDidMount() {
    if(this.input) {this.input.focus();}
  }

  render() {
    const {
      value,
      onChange,
      onSubmit,
      children
    } = this.props;

    return (
      <form onSubmit={onSubmit} style={{display: 'flex', flexDirection:'row', flexWrap: 'wrap',
      justifyContent: 'space-around'}}>
        <TextField
          label="Search Anything!"
          style={{margin: '0 2em', width: '15em'}}
          placeholder="Redux"
          value={value}
          onChange={onChange}
          inputRef={(node) => { this.input = node; }}
        />
        <RaisedButton size='large' color="primary" variant="raised" type="submit" style={{margin: '1em 0'}}>
          {children}
        </RaisedButton>
      </form>
    );
  }
}

Search.propTypes = {
  value: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired,
};

export default Search;
