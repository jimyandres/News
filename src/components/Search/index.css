.inputText {
  margin-right: 1em;
}
