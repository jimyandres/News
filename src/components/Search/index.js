import React, { Component } from 'react';
import PropTypes from 'prop-types';
import RaisedButton from 'material-ui/RaisedButton';
import TextField from 'material-ui/TextField';
import './index.css';

class Search extends Component {
  componentDidMount() {
    if(this.input) {this.input.focus();}
  }

  render() {
    const {
      value,
      onChange,
      onSubmit,
      children
    } = this.props;

    return (
      <form onSubmit={onSubmit}>
        <TextField
          floatingLabelText="Search anything!"
          type="text"
          value={value}
          onChange={onChange}
          ref={(node) => { this.input = node; }}
          className='inputText'
        />
        <RaisedButton type="submit" primary label={children} />
      </form>
    );
  }
}

Search.propTypes = {
  value: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired,
};

export default Search;
