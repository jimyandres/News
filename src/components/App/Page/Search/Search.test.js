import React from 'react';
import ReactDOM from 'react-dom';
import { configure,shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallowToJson } from 'enzyme-to-json';
import Search from './index';

configure({ adapter: new Adapter() });

describe('Search', () => {

	const props = {
		onChange: jest.fn(),
		onSubmit: jest.fn(),
	};

	it('render without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(<Search {...props} >Search</Search>, div);
	});

	test('has a valid snapshot', () => {
		const wrapper = shallow(<Search {...props} >Search</Search>);
		expect(shallowToJson(wrapper)).toMatchSnapshot();
	});
});
