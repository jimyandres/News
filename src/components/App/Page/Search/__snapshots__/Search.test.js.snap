// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search has a valid snapshot 1`] = `
<form
  onSubmit={[Function]}
  style={
    Object {
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "justifyContent": "space-around",
    }
  }
>
  <TextField
    inputRef={[Function]}
    label="Search Anything!"
    onChange={[Function]}
    placeholder="Redux"
    required={false}
    select={false}
    style={
      Object {
        "margin": "0 2em",
        "width": "15em",
      }
    }
  />
  <WithStyles(Button)
    color="primary"
    size="large"
    style={
      Object {
        "margin": "1em 0",
      }
    }
    type="submit"
    variant="raised"
  >
    Search
  </WithStyles(Button)>
</form>
`;
